src/vs/editor/contrib/rename/browser/rename.ts

```diff
@@ -117,34 +144,64 @@ class RenameController implements IEditorContribution {
 		return RenameController.ID;
 	}
 
-	public run(): TPromise<void> {
+	public async run(): TPromise<void> {
+		const selection = this.editor.getSelection();
 
-		const selection = this.editor.getSelection(),
-			word = this.editor.getModel().getWordAtPosition(selection.getStartPosition());
+		let lineNumber = selection.startLineNumber,
+			selectionStart = 0,
+			selectionEnd = 0,
+			wordRange: Range,
+			word: string;
+
+		let initialValue = await resolveInitialRenameValue(this.editor.getModel(), this.editor.getPosition());
+
+		if(initialValue) {
```

Comment:
We can maybe move this to a 'default' implementation of resolveRenameValue

Reply:
I'm not really sure, the code below is just setting values local to the `run` function. It's only small extension of the logic that was already there before.